### ISSUE TITLE
'no-unused-vars' flag despite being used in a template string

### ISSUE THREAD

**Issue by Shockolate** [eslint/eslint#6143]
**What version of ESLint are you using?**
v2.9.0

**What parser (default, Babel-ESLint, etc.) are you using?**
The default, I believe.

**Please show your full configuration:**
```yaml
extends: airbnb
env:
  node: true
  mocha: true
  es6: true
rules:
  consistent-return: off
  strict: off
  no-console: off
```
**What did you do? Please include the actual source code causing the issue.**
I'm new to Javascript, so don't judge my code too harshly...
```javascript
function sqlExecute(nameAndSqlParams, callback) {
  return sql.execute(nameAndSqlParams.seriateParams).then(
    (data) => (callback(null, data)),
    (err) => (callback('Error ${nameAndSqlParams.name} : ${err}'))
  );
}
```

**What did you expect to happen?**
No error or flag should be thrown with the no-unused-vars set to on.

**What actually happened? Please include the actual, raw output from ESLint.**
```
$ ./node_modules/eslint/bin/eslint.js src/Persistence.js
   10:6   error  'err' is defined but never used      no-unused-vars
✖ 4 problems (4 errors, 0 warnings)
```

**Comment by ben-page** [comment]
The problem is that your string isn't a template string. Template strings use backticks (`). Your string uses single quotes (').

**Comment by ben-page** [comment]
```
function sqlExecute(nameAndSqlParams, callback) {
  return sql.execute(nameAndSqlParams.seriateParams).then(
    (data) => (callback(null, data)),
    (err) => (callback(`Error ${nameAndSqlParams.name} : ${err}`))
  );
}
```

**Comment by Shockolate** [comment]
The flag was no longer thrown when using the Grave Accent.
I didn't think to use the Grave Accent for my application was working with the Single Quote.
Closing
